server/plugin/api_test.go

```diff
@@ -1088,6 +1088,67 @@ func TestCreateSubscription(t *testing.T) {
 			},
 			ExpectedStatusCode: http.StatusCreated,
 		},
+		"success with subscription type bulk": {
+			RequestBody: fmt.Sprintf(`{
+				"user_id": "%s",
+				"type": "%s",
+				"channel_id": "%s",
+				"filters": "mockFilters"
+			  	}`, testutils.GetID(), constants.SubscriptionTypeBulk, testutils.GetChannelID()),
+			SetupAPI: func(api *plugintest.API) {},
+			SetupClient: func(client *mock_plugin.Client) {
+				client.On("GetTableFieldsFromServiceNow", mock.AnythingOfType("string")).Return(
+					testutils.GetServiceNowTableFields(3), http.StatusOK, nil,
+				)
+
+				client.On("CheckForDuplicateSubscription", mock.AnythingOfType("*serializer.SubscriptionPayload")).Return(
+					false, http.StatusOK, nil,
+				)
+
+				client.On("CreateSubscription", mock.AnythingOfType("*serializer.SubscriptionPayload")).Return(
+					http.StatusCreated, nil,
+				)
+			},
+			SetupPlugin: func(p *Plugin) {
+				var s *serializer.SubscriptionPayload
+				monkey.PatchInstanceMethod(reflect.TypeOf(s), "IsValidForCreation", func(_ *serializer.SubscriptionPayload, _ string) error {
+					return nil
+				})
+
+				monkey.PatchInstanceMethod(reflect.TypeOf(p), "HasChannelPermissions", func(_ *Plugin, _, _ string) (int, error) {
+					return http.StatusOK, nil
+				})
+			},
+			ExpectedStatusCode: http.StatusCreated,
+		},
+		"error with subscription type bulk": {
+			RequestBody: fmt.Sprintf(`{
+				"user_id": "%s",
```

Comment:
create a mock function for this request body with param record or bulk subscription and use that here